ts/components/dialog/user-settings/pages/AppearanceSettingsPage.tsx

```diff
@@ -0,0 +1,467 @@
+import { LUCIDE_ICONS_UNICODE } from '../../../icon/lucide';
+import { LucideIcon } from '../../../icon/LucideIcon';
+import { H9 } from '../../../basic/Heading';
+import { getMenuAnimation } from '../../../menu/MenuAnimation';
+import { ItemWithDataTestId } from '../../../menu/items/MenuItemWithDataTestId';
+import { ZOOM_FACTOR } from '../../../../session/constants';
+import { SessionContextMenuContainerItemsCentered } from '../../../SessionContextMenuContainer';
+
+const StyledPrimaryColorSwitcherContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+  padding-inline: 20px;
+  align-items: center;
+  align-self: center;
+  width: 100%;
+  justify-content: space-around;
+  height: var(--panel-button-container-min-height);
+`;
+
+function PrimaryColorSwitcher() {
+  const selectedPrimaryColor = usePrimaryColor();
+  const dispatch = useDispatch();
+  const diameterRadioBorder = 35;
+
+  return (
+    <StyledPrimaryColorSwitcherContainer>
+      {getPrimaryColors().map(item => {
+        const overriddenColorsVars = {
+          '--primary-color': item.color,
+          '--text-primary-color': item.id === selectedPrimaryColor ? undefined : 'transparent',
+        } as React.CSSProperties;
+        return (
+          <RadioDot
+            selected={true}
+            onClick={() => {
+              void switchPrimaryColorTo(item.id, dispatch);
+            }}
+            key={item.id}
+            disabled={false}
+            dataTestId={undefined}
+            diameterRadioBorder={diameterRadioBorder}
+            style={overriddenColorsVars}
+            ariaLabel={item.ariaLabel}
+          />
+        );
+      })}
+    </StyledPrimaryColorSwitcherContainer>
+  );
+}
+
+const StyledPreview = styled.svg`
+  flex-shrink: 0;
+  height: 58px;
+`;
+
+const ThemeIcon = (props: { style: StyleSessionSwitcher }) => {
+  return (
+    <StyledPreview xmlSpace="preserve" viewBox="0 0 80 72" fill={props.style.background}>
+      <path
+        stroke={props.style.border}
+        d="M7.5.9h64.6c3.6 0 6.5 2.9 6.5 6.5v56.9c0 3.6-2.9 6.5-6.5 6.5H7.5c-3.6 0-6.5-2.9-6.5-6.5V7.4C1 3.9 3.9.9 7.5.9z"
+      />
+      <path
+        fill={props.style.receivedBackground}
+        d="M8.7 27.9c0-3.2 2.6-5.7 5.7-5.7h30.4c3.2 0 5.7 2.6 5.7 5.7 0 3.2-2.6 5.7-5.7 5.7H14.4c-3.1.1-5.7-2.5-5.7-5.7z"
+      />
+      <path
+        fill={props.style.sentBackground}
+        d="M32.6 42.2c0-3.2 2.6-5.7 5.7-5.7h27c3.2 0 5.7 2.6 5.7 5.7 0 3.2-2.6 5.7-5.7 5.7h-27c-3.1 0-5.7-2.5-5.7-5.7z"
+      />
+    </StyledPreview>
+  );
+};
+
+const StyledThemeName = styled.div`
+  font-size: var(--font-size-h8);
+  color: var(--text-primary-color);
+  font-weight: 700;
+  flex-shrink: 0;
+`;
+
+const Themes = () => {
+  const themes = getThemeColors();
+  const selectedTheme = useTheme();
+  const dispatch = useDispatch();
+
+  return (
+    <>
+      {themes.map(theme => (
+        <>
+          <PanelButton
+            dataTestId="invalid-data-testid"
```

Comment:
Should this be invalid?

Reply:
hmm thats tricky, not actually sure whats best to do here. having `dataTestId="invalid-data-testid"` everywhere feels like a rough solution, but it is good to catch when we miss it. Happy to keep for now, not sure what we should do going forward